File: CPP/Leetcode/14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& nums) {
        
    string res="";
       for(int i=0;i<nums[0].size();i++)
       {
           for(int j=0;j<nums.size()-1;j++)
           {
               if(nums[j][i]!=nums[j+1][i])
                   return res;
           }
           res=res+nums[0][i];
           
       }
        return res;
        
    }
};

File: Some Leetcode and GFG Solutions/14-longest-common-prefix/14-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        lil= min(strs,key=len)
        for i,j in enumerate(lil):
            
#             i= int, j=char
            for k in strs:
                if k[i]!=j:
                    return lil[:i]
#                 either a string has no. of same chars lower than lil or equal to lil itself 
        return lil